deepchecks/tabular/checks/integrity/outlier_sample_detection.py

```diff
@@ -0,0 +1,160 @@
+    Parameters
+    ----------
+    columns : Union[Hashable, List[Hashable]] , default: None
+        Columns to check, if none are given checks all columns except ignored ones.
+    ignore_columns : Union[Hashable, List[Hashable]] , default: None
+        Columns to ignore, if none given checks based on columns variable
+    num_nearest_neighbors : int, default: 10
+        Number of nearest neighbors to use for outlier detection.
+    extent_parameter: int, default: 3
+        Extend parameter for LoOP algorithm.
+    n_samples : int , default: 50_000
+        number of samples to use for this check.
+    n_to_show : int , default: 5
+        number of data elements with the highest outlier score to show (out of sample).
+    random_state : int, default: 42
+        random seed for all check internals.
+    """
+
+    def __init__(
+            self,
+            columns: Union[Hashable, List[Hashable], None] = None,
+            ignore_columns: Union[Hashable, List[Hashable], None] = None,
+            num_nearest_neighbors: int = 10,
+            extent_parameter: int = 3,
+            n_samples: int = 50_000,
+            n_to_show: int = 5,
+            random_state: int = 42,
+            **kwargs
+    ):
+        super().__init__(**kwargs)
+        if not isinstance(extent_parameter, int) or extent_parameter <= 0:
+            raise DeepchecksValueError('extend_parameter must be a positive integer')
+        if not isinstance(num_nearest_neighbors, int) or num_nearest_neighbors <= 0:
+            raise DeepchecksValueError('num_nearest_neighbors must be a positive integer')
+        self.columns = columns
+        self.ignore_columns = ignore_columns
+        self.num_nearest_neighbors = num_nearest_neighbors
+        self.extent_parameter = extent_parameter
+        self.n_samples = n_samples
+        self.n_to_show = n_to_show
+        self.random_state = random_state
+
+    def run_logic(self, context: Context, dataset_type: str = 'train') -> CheckResult:
+        """Run check."""
+        if dataset_type == 'train':
+            dataset = context.train
+        else:
+            dataset = context.test
+        dataset = dataset.sample(self.n_samples, random_state=self.random_state, drop_na_label=True)
+        df = select_from_dataframe(dataset.data, self.columns, self.ignore_columns)
+        if self.num_nearest_neighbors >= len(df):
+            logger.warning('Passed num_nearest_neighbors is greater than the number of samples in the dataset')
+            self.num_nearest_neighbors = len(df) - 1
+
+        # Calculate distances matrix and retrieve nearest neighbors based on distance matrix.
+        df_cols_for_gower = df[dataset.cat_features + dataset.numerical_features]
+        is_categorical_arr = np.array(df_cols_for_gower.columns.map(lambda x: x in dataset.cat_features), dtype=bool)
+        dist_matrix, idx_matrix = gower_distance.gower_matrix_n_closets(data=np.asarray(df_cols_for_gower),
```

Comment:
Bonus points - catching the out of memory exception that may raise if the user by mistake sets num_nearest_neighbors to be same order of magnitude as len(df) and raise instead a deepchecks error telling him to select a lower value.